packages/data-context/src/sources/FileDataSource.ts

```diff
@@ -72,7 +72,15 @@ export class FileDataSource {
 
       return files
     } catch (e) {
-      debug('error in getFilesByGlob %o', e)
+      if (!globOptions.suppressErrors) {
+        // Log error and retry with filesystem errors suppressed - this allows us to find partial
+        // results even if the glob search hits permission issues (#24109)
+        debug('Error in getFilesByGlob %o, retrying with filesystem errors suppressed', e)
+
+        return await this.getFilesByGlob(cwd, glob, { ...globOptions, suppressErrors: true })
```

Comment:
Should we just do this by default instead of the "if it fails, retry and suppress errors"?

Reply:
That was my original plan, but if we do that we get _zero_ logging of the problem. My fear is that a user will have a spec file within a restricted directory and that spec will just never show up and there will be no logging (debug or otherwise) explaining why which will make it very difficult to troubleshoot.

Taking the "retry with suppress" approach is less performant, but gives us the possibility of debug logs.